components/public-api/gitpod/v1/scm.proto

```diff
@@ -0,0 +1,65 @@
+syntax = "proto3";
+
+package gitpod.v1;
+
+import "google/protobuf/timestamp.proto";
+
+option go_package = "github.com/gitpod-io/gitpod/components/public-api/go/v1";
+
+service SCMService {
+  rpc GetToken(GetTokenRequest) returns (GetTokenResponse) {}
+  rpc GuessTokenScopes(GuessTokenScopesRequest) returns (GuessTokenScopesResponse) {}
+  rpc SearchRepositories(SearchRepositoriesRequest) returns (SearchRepositoriesResponse) {}
+  rpc ListSuggestedRepositories(ListSuggestedRepositoriesRequest) returns (ListSuggestedRepositoriesResponse) {}
+}
+
+message GetTokenRequest {
+    string host = 1;
+    string organization_id = 2;
+}
+message GetTokenResponse {
+    GitToken token = 1;
+}
+
+message GuessTokenScopesRequest {
+    string host = 1;
+    string repo_url = 2;
+    string git_command = 3;
+}
+message GuessTokenScopesResponse {
+    repeated string scopes = 1;
+    string message = 2;
+}
+
+message SearchRepositoriesRequest {
+    string organization_id = 1;
+    string search_string = 2;
+    int32 limit = 3;
```

Comment:
it should be pagination request

Reply:
See, the search is delegating to 3rd party services, where at most we can limit the results by number. We need to abstract over several APIs actually, and implementing quirks with pagination seems to be absolutely unhelpful.